src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -1597,3 +1599,37 @@ cmp arr1 arr2 =
 instance NFData (Array a) where
     {-# INLINE rnf #-}
     rnf Array {} = ()
+
+strip :: forall a m. (Storable a, MonadIO m) =>
+    (a -> Bool) -> Array a -> m (Array a)
+strip eq arr@Array{..} =
+    let p = unsafeForeignPtrToPtr aStart
+        q = aEnd
+        len = length arr
+        st = getStart p len
+        end = do
+            st' <- st
+            if st' == q
+            then return q
+            else getLast $ q `plusPtr` negate (sizeOf (undefined :: a))
+    in go st end
```

Comment:
`go` is not a recursive function. We can inline the definition right here, that would be more readable.